manager/manager.go

```diff
@@ -746,6 +746,9 @@ func (m *manager) registerCollectors(collectorConfigs map[string]string, cont *c
 
 // Create a container.
 func (m *manager) createContainer(containerName string) error {
+	m.containersLock.Lock()
```

Comment:
Now that you moved out the creation of goroutines from the initialization of container handlers, is it necessary to guard the entire function with the lock?